src/main/resources/reference.conf

```diff
@@ -1,106 +1,246 @@
-        },
-        workspace-creator = {
-          roleActions = ["create_workspace", "share_policy::can-compute-user", "read_policy::can-compute-user"]
-        },
-        batch-compute-user = {
-          roleActions = ["launch_batch_compute"]
-        },
-        notebook-user = {
-          roleActions = ["launch_notebook_cluster"]
-        }
-      }
-      reuseIds = true
-    },
-    notebook-cluster = {
-      actionPatterns = ["status", "connect", "sync", "delete", "read_policies", "stop_start"]
-      ownerRoleName = "creator"
-      roles = {
-        creator = {
-          roleActions = ["status", "connect", "sync", "delete", "read_policies", "stop_start"]
-        }
-      }
-      reuseIds = true
-    },
-    cloud-extension = {
-      actionPatterns = ["get_pet_private_key", "alter_policies", "read_policies"]
-      ownerRoleName = "owner"
-      roles = {
-        owner = {
-          roleActions = ["alter_policies", "read_policies"]
-        },
-        google = {
-          roleActions = ["get_pet_private_key"]
-        }
-      }
-      reuseIds = false
+      }
+    }
+    reuseIds = false
+  }
+  billing-project = {
+    actionPatterns = {
+      create_workspace = {
+        description = "Create new workspaces in this billing-project"
+      }
+      alter_policies = {
+        description = "Create and delete policies for this billing-project"
+      }
+      read_policies = {
+        description = "List all policies and policy details for this billing-project"
+      }
+      launch_batch_compute = {
+        description = ""
+      }
+      list_notebook_cluster = {
+        description = ""
+      }
+      launch_notebook_cluster = {
+        description = ""
+      }
+      sync_notebook_cluster = {
+        description = ""
+      }
+      delete_notebook_cluster = {
+        description = ""
+      }
+      stop_start_notebook_cluster = {
+        description = ""
+      }
+      alter_google_role = {
+        description = ""
+      }
+      "share_policy::.+" = {
+        description = ""
+      }
+      "read_policy::.+" = {
+        description = ""
+      }
+    }
+    ownerRoleName = "owner"
+    roles = {
+      owner = {
+        roleActions = ["create_workspace", "alter_policies", "read_policies", "launch_batch_compute", "list_notebook_cluster", "launch_notebook_cluster", "sync_notebook_cluster", "delete_notebook_cluster", "alter_google_role"]
+      },
+      workspace-creator = {
+        roleActions = ["create_workspace", "share_policy::can-compute-user", "read_policy::can-compute-user"]
+      },
+      batch-compute-user = {
+        roleActions = ["launch_batch_compute"]
+      },
+      notebook-user = {
+        roleActions = ["launch_notebook_cluster"]
+      }
+    }
+    reuseIds = true
+  }
+  notebook-cluster = {
+    actionPatterns = {
```

Comment:
@rtitle given that these are actions that your team cares about, would you (or anyone else from your team) like to do the honors of describing them? there's also some notebook actions under `billing-project`

they would need to be written such that the sentence `s"Users who can perform ${action.name} may ${action.description}".` could be generated

otherwise, we'll do our best to come up with the descriptions

Reply:
Sorry I missed this earlier. Here's an attempt:

status = "view notebook cluster status details and configuration"
connect = "connect to the Jupyter notebook running on the notebook cluster"
sync = "sync files to/from the notebook cluster VMs"
delete = "delete the notebook cluster"
read_policies = "view all policies and policy details for the notebook cluster"
stop_start = "stop and start the notebook cluster VMs"